docs/reference/deprecation-notice.md

```diff
@@ -0,0 +1,76 @@
+# Deprecation Notice
+---
+
+This document outlines the features that will be deprecated in the next major release v2.0.0. Deprecation means that these features will no longer be supported or recommended for use. Users should transition to alternative features or solutions as specified below.
```

Comment:
It this for upcoming v2.0.0 only? Did we have any deprecation in v1.x?

Reply:
Synced offline with @susanshi, I'll have deprecation version and removal version for features. The listed features are deprecated in v1.x, and will be completely removed in v2.0